### ISSUE TITLE
[MAUI-Windows] Unhandled layout cycle exception occurs when setting "IsVisible = False" inside FlexLayout

### ISSUE THREAD

**Issue by Deepak1799** [dotnet/maui#9377]
### Description

I am trying to add a view (say Grid) inside a FlexLayout. After added the view, when I set "IsVisible" property of the grid at the run time. I am facing the below unhandled exception 

**Exception:** "A cycle occurred while laying out the GUI."
**Message:** "Layout cycle detected.  Layout could not complete."

<img width="701" alt="image" src="https://user-images.githubusercontent.com/65169077/184359682-1e36004c-d5ba-4478-a5ee-eb52af05b934.png">

**Note:** This occurs with all kind of MAUI views inside a FlexLayout.

### Steps to Reproduce

1. Create a new MAUI App.
2. Add a view (say grid) inside the flex layout.
3. Use a switch control to to toggle the visibility of the view at the run time.
4. Run the sample.
5. Disable the view at the run time by setting its property "IsVisible=false". Now the unhandled Exception will be thrown.

_**I have attached a sample to reproduce the issue.**_
[MauiApp1.zip](https://github.com/dotnet/maui/files/9316478/MauiApp1.zip)

Run the attached sample and toggle the switch control "Make Visible", in the left top of the app.

### Version with bug

6.0.419

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10.0.19041

### Did you find any workaround?

No

### Relevant log output

```shell
No logs obtained due to unhandled exception occurs.
```

**Comment by AppXture** [comment]
I'm having similar issues that started today after updating VS 2022 to 17.3 and VS2022 Preview to 17.4. 

My steps to reproduce is that I have a FlexLayout on a ContentPage and if I add anything to it then I get the error. If I comment out the  FlexLayout or it's contents then no error.

It's been working fine until the latest Visual Studio updates. Not sure which version of Windows SDK I was on before but I'm on 10.0.19041 now.

**Comment by SunshineSpring666** [comment]
Hi guys, 
Any workaround found? Same issue encountered here. 
FlexLayout is crucial in wrapping things based on dynamic window sizes. 
Wonder if there are any updates about this issue?

**Comment by holecekp** [comment]
I am experiencing a similar crash in FlexLayout. It worked perfectly in the Xamarin.Forms. The MAUI version of the same code crashes on Windows (layout cycle exception).

**Comment by SunshineSpring666** [comment]
This bug still exists in latest MAUI with .NET 7 RC.
There are many scenarios in which the UI components need be adjusted dynamically according to window width change, and FlexLayout's wrap is the proper way. 
Hope this could be solved sooner in its upcoming servicing update.

**Comment by mikeedwards83** [comment]
Not sure how this is being moved to backlog. Flexlayout is completely unusable at the moment.

**Comment by hartez** [comment]
I just tried the repro app with the latest stable VS and .NET 7, and it appears to be working fine:

![2022-11-22_17-11-57](https://user-images.githubusercontent.com/538025/203446109-c0843318-18b0-4f03-8ce9-c2030c265eb9.gif)

**Comment by holecekp** [comment]
I have installed the latest 7.0.49 version and my app is still crashing as before. I first thought that the bug has not been fixed. But then I noticed that the error message is different and that it crashes after the update even if I remove the flex layout. I have  downloaded the repro project from this issue, and tried it with the latest .net6 and .net7 version, and it works without crashing.

So yes, this bug has been fixed but it was replaced by another one that crashes the app in different more general situations. If I have time, I will try to find the problem, create a repro project and file a new issue for it. It seems that MAUI on .NET 7 have problem with some lazily-loaded views (even without flex layout) that used to work in Xamarin.Forms and older versions of MAUI. When a lazily-loaded view should be loaded and the LazyView.Content is set to the newly created view, the app crashes:
```
Exception thrown: 'System.InvalidOperationException' in Microsoft.Maui.dll
The handler's MauiContext cannot be null.
```
This is a Windows-only problem. It works on Android. I will investigate it more and file a new issue.
